### DISCUSSION TITLE
Refactor build command to not create the docker image

### DISCUSSION THREAD

**Discussion by cdwillie76** [Zuehlke/poinz#316] [Ideas]
In order to build and deploy PoinZ in our internal environment, I need to build PoinZ inside a container (vs. my local system).  Here is what I am doing in my local PoinZ repo
- I comment out the fifth step from the dockerImage.mjs
- I created a new Dockerfile (using a multi-stage build), call npm install and build, added the steps from the original Dockerfile, and adds all the other bits I need (different base image, configure this, disable that, etc).  

While this works, it isn't as clean as I would like. In a future release, could the build command be split so there is a command to just build and another to create the docker image? Also, if there is a better workflow for this, I am open to learning (new to docker and node).

**Comment by xeronimus** [comment]
Hi @cdwillie76,


we could split up the build process as follows:

```$ npm run build``` would do the current steps 1-4. This will leave us with a **/deploy** directory, ready to be copied to a server with node runtime.

```$ npm run build:dockerimage``` would first call ```npm run build``` and then do step 5.

Would that solve your problem, or help in any way?

**Comment by xeronimus** [comment]
I quickly implemented the change as a draft on branch 317
You can checkout branch [317](https://github.com/Zuehlke/poinz/tree/317-split-build-process) and test it, if you want :)

**Reply by cdwillie76** [reply]
Thanks, I will give it a try today!

**Reply by cdwillie76** [reply]
That worked for me.  Here is a Dockerfile that builds poinz in the container before creating the deployment layer - https://github.com/cdwillie76/poinz/blob/317-split-build-process/Dockerfile.XX.  The standalone build command will help reduce the changes I need to make when a new poinz version is released to get it built to deploy internally. 

Thanks for working on this change!

**Reply by xeronimus** [reply]
do we need to change / add something else on branch 317? or can we merge this to master?

**Reply by cdwillie76** [reply]
I don't see anything else that needs to be added/changed and it can be merged.  Thanks!
